gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/util/CollectionUtil.java

```diff
@@ -74,4 +96,22 @@ public static <K,V> ConcurrentHashMap<K,V> clone(final ConcurrentHashMap<K,V> ma
 
         return result;
     }
+
+    /**
+     * Adds the element argument to the start of the supplied array argument, but if the array is null then return the
+     * element as a new array with just that item in it.
+     */
+    public static <T> T[] addFirst(final T[] array, final T element, final Class<T> clazz) {
+        if (null == array && null == element) {
```

Comment:
if element is null then
`singleElementArray[0] = null;` and `singleElementArray[0] = null;` is the same
so this `if` branch not needed